engine/tests/kernel_tests/window_overlap_kernel_test.cpp

```diff
@@ -1293,33 +1409,417 @@ TEST_F(WindowOverlapAccumulatorTest, BigWindowSingleNode) {
+            overlap_accumulator_kernel,
+            context);
+
+    // run function in overlap accumulator
+    std::thread run_thread_accumulator = std::thread([overlap_accumulator_kernel](){
+        kstatus process = overlap_accumulator_kernel->run();
+        EXPECT_EQ(kstatus::proceed, process);
+    });
+
+    std::this_thread::sleep_for(std::chrono::milliseconds(10));
+
+    // add data into the Accumulator CacheMachines
+    for (std::size_t i = 0; i < batchesCacheData.size(); i++) {
+        batchesCacheMachineAccumulator->addCacheData(std::move(batchesCacheData[i]));
+        if (i != 0) {
+            std::this_thread::sleep_for(std::chrono::milliseconds(2));
+            precedingCacheMachineAccumulator->addCacheData(std::move(batchesPrecedingCacheData[i - 1]));
+        }
+        if (i != batchesCacheData.size() - 1) {
+            std::this_thread::sleep_for(std::chrono::milliseconds(2));
+            followingCacheMachineAccumulator->addCacheData(std::move(batchesFollowingCacheData[i]));
+        }
+        std::this_thread::sleep_for(std::chrono::milliseconds(10));
+    }
+    batchesCacheMachineAccumulator->finish();
+    precedingCacheMachineAccumulator->finish();
+    followingCacheMachineAccumulator->finish();
+
+    run_thread_accumulator.join();
+
+    // get and validate output
+    return outputCacheMachineAccumulator->pull_all_cache_data();
+}
+
+TEST_F(WindowOverlapGeneratorTest, BasicSingleNode) {
+
+    size_t size = 100000;
+
+    // Define full dataset
+    auto iter0 = cudf::detail::make_counting_transform_iterator(0, [](auto i) { return int32_t(i);});
+    auto iter1 = cudf::detail::make_counting_transform_iterator(1000, [](auto i) { return int32_t(i * 2);});
+    auto iter2 = cudf::detail::make_counting_transform_iterator(0, [](auto i) { return int32_t(i % 5);});
+    auto valids_iter = cudf::detail::make_counting_transform_iterator(0, [](auto /*i*/) { return true; });
+
+    cudf::test::fixed_width_column_wrapper<int32_t> col0(iter0, iter0 + size, valids_iter);
+    cudf::test::fixed_width_column_wrapper<int32_t> col1(iter1, iter1 + size, valids_iter);
+    cudf::test::fixed_width_column_wrapper<int32_t> col2(iter2, iter2 + size, valids_iter);
+
+    CudfTableView full_data_cudf_view ({col0, col1, col2});
+
+    int preceding_value = 50;
+    int following_value = 10;
+    std::vector<cudf::size_type> batch_sizes = {5000, 50000, 20000, 15000, 10000}; // need to sum up to size
+
+    // define how its broken up into batches and overlaps
+    std::vector<std::string> names({"A", "B", "C"});
+    std::vector<std::unique_ptr<CacheData>> preceding_overlaps, batches, following_overlaps;
+    std::vector<std::unique_ptr<CacheData>>& inputCacheData = batches;
+    std::tie(preceding_overlaps, batches, following_overlaps) = break_up_full_data(full_data_cudf_view, preceding_value, following_value, batch_sizes, names);
```

Comment:
for the overlap generator, `break_up_full_data` is actually giving you the both the input you need and all the expected outputs.
The input is `batches`
The expected outputs are preceding_overlaps, batches, following_overlaps.

WHich means you dont need `make_expected_generator_output`

Reply:
If necessary because `batches` is the input, and `inputCacheMachineGenerator->addCacheData(std::move(batches[i]))` prevents reuse of the same object. So create another one with `make_expected_generator_output`